compiler-cli/src/new.rs

```diff
@@ -526,6 +542,44 @@ pub fn hello_world_test() {{
             ),
         )
     }
+
+    fn docker_compose(&self) -> Result<()> {
+        match self.options.docker {
+            Some(choice) => {
+                let docker_type = match choice.unwrap_or(Docker::Default) {
+                    Docker::Default => "",
+                    Docker::Slim => "-slim",
+                    Docker::Alpine => "-alpine",
+                };
+                write(
+                    self.root.join("docker-compose.yml"),
+                    &format!(
+                        r#"version: "3"
+    
+services:
+  {name}:
+    image: ghcr.io/gleam-lang/gleam:v{version}-erlang{docker_type}
```

Comment:
Note this will not work for people building gleam from master because the version is set to `x.y.z-dev` and there is no corresponding docker image.

I think this is an ok trade-off because people willing to work of main are probably ok changing this value.

Reply:
Second note this always assumes you want a  erlang version because projects generated by a gleam new are rebar3 projects. 

There might be a future where you generate new mix/npm projects in which case when adding those you would want to choose the approprate image. However because of native gleam tooling there might not be a future where we generate mix/npm projects so for now assuming erlang seems sensible